DietPi-Software | Aria2: Installer rework

msongz opened the pull request:
to avoid losing tasks after reboot the device

<!--
Before submitting a pull request:
- Please ensure the target branch is "dev" (active development): https://github.com/MichaIng/DietPi/tree/dev
- Please ensure changes have been tested and verified functional.
-->
**Status**: Testing
- [x] should adding /var/lib/dietpi/dietpi-software/installed/aria2.session first to make it works?

msongz committed:
Update gitea to 1.8.2

msongz committed:
change gitea to 1.8 for stable purpose

Fourdee committed:
Pateron Legend addition

oct8l

msongz committed:
Merge pull request #1 from MichaIng/master

update from source

msongz committed:
update gitea to 1.9.3

msongz committed:
add SAVE SESSION to aria2.conf

to avoid losing any tasks after reboot the device
and remove duplicate lines

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -11139,9 +11144,7 @@ bt-max-peers=$(Optimise_BitTorrent 2)
-max-overall-upload-limit=0
 max-upload-limit=0
-max-overall-download-limit=0
 max-download-limit=0
```
@msongz 
Why you removed the overall limits? They default to `0` anyway, but this is true for the per-torrent limits as well.
#### EDIT: Ah I see the doubled entries.

msongz replied:
😄

MichaIng commented:
@msongz 
Many thanks for this, looks very reasonable.
What do you think about moving the aria2.conf and session files to `/mnt/dietpi_userdata`? E.g. in other downloaders we save the session file right in `/mnt/dietpi_userdata/downloads/`:
- qBittorrent save and tmp paths: https://github.com/MichaIng/DietPi/blob/dev/dietpi/dietpi-software#L10286-L10288
- ruTorrent session dir: https://github.com/MichaIng/DietPi/blob/dev/dietpi/dietpi-software#L10541
- and others as well.

E.g.
```
/mnt/dietpi_userdata/aria2/aria2.conf
/mnt/dietpi_userdata/downloads/aria2.session
```
Or both in `/aria2/`, not sure what is best to go with here, however I like the idea to keep configs and states in userdata, since this can be easily moved to external drive and is there safer against data corruption/broken SDcard and can be easily migrated to a new system.

> should be add a file : `/var/lib/dietpi/dietpi-software/installed/aria2.session` first to make it works?

Did you test if there are any errors when starting aria2c with `input-file` pointing to non-existent file? Probably a warning, but I assume it would simply start and then create the file on first periodic save interval.

msongz commented:
`Exception: [download_helper.cc:563] errorCode=1 Failed to open the file /var/lib/dietpi/dietpi-software/installed/aria2.session, cause: File not found or it is a directory`
after I test update the config with the session file and restart the service with systemctl, it prompts an error  and prevents the service start.
so the session file may be created first to make this works

msongz commented:
regarding the config and the session file path, I love your point

MichaIng commented:
@msongz 
Perfect, I work on quite some rework now, also to run aria2 as user "aria2", group "dietpi" and umask 002, so media player run users within dietpi group can directly access aria2 downloads.

MichaIng committed:
v6.26

+ DietPi-Software | Aria2: Runs now as its own user "aria2" with primary group "dietpi" and creates downloads with umask 002. Besides fulfilling common security standards, this allows other download tools and media players, installed via dietpi-software, to access those files without the need to manually adjust permissions.
+ DietPi-Software | Aria2: aria2.conf has been moved to /mnt/dietpi_userdata/aria2/, the new aria2 users home dir.

MichaIng commented:
@msongz 
Okay, changes done. What you think about it?

Requires some patch on update to create new user and move config file.

msongz commented:
nice work! would like to test it ASAP. 😄

MichaIng commented:
@msongz 
Okay will merge for testing, especially since your branch is based on master while we should test with current dev branch around 😉.

#### EDIT: Whoopsie, dumb syntax/forgotten username 😄: https://github.com/MichaIng/DietPi/commit/1a5d29d9713c133dafddebdaa8745e21153b7c2b
#### EDIT2: Log file needs to be pre-created: https://github.com/MichaIng/DietPi/commit/7573a511156021cafcce5d6334b6561477af8f53

Okay after above commits it works well here now, fresh installs and reinstalls.

Changelog: https://github.com/MichaIng/DietPi/commit/1b9597399ab5a671e808170dbd3603ee857c8ed1

On v6.26 update, apply new session store/resume feature but preserve custom user settings, if applied: https://github.com/MichaIng/DietPi/commit/58455252a7a81dd0de1894651dbd7201b78d1920

Another fix for web UI install:
- New dir was moved **into** the existing dir, thus nothing was updated.
- Now existing dir is fully removed, since those files are never written to, settings are stored in browser/client only: https://github.com/MichaIng/DietPi/commit/44ce25bc7b55197e159bcf7f62a048ad6b886d52
